pkg/lint/lintersdb/manager.go

```diff
@@ -992,6 +956,42 @@ func (m Manager) GetAllLinterConfigsForPreset(p string) []*linter.Config {
 	return ret
 }
 
+func linterConfigsToMap(lcs []*linter.Config) map[string]*linter.Config {
+	ret := map[string]*linter.Config{}
+	for _, lc := range lcs {
+		lc := lc // local copy
+		ret[lc.Name()] = lc
+	}
+
+	return ret
+}
+
+func AllPresets() []string {
+	return []string{
+		linter.PresetBugs,
+		linter.PresetComment,
+		linter.PresetComplexity,
+		linter.PresetError,
+		linter.PresetFormatting,
+		linter.PresetImport,
+		linter.PresetMetaLinter,
+		linter.PresetModule,
+		linter.PresetPerformance,
+		linter.PresetSQL,
+		linter.PresetStyle,
+		linter.PresetTest,
+		linter.PresetUnused,
+	}
+}
+
+func allPresetsSet() map[string]bool {
+	ret := map[string]bool{}
```

Comment:
I think this method is mainly useless because it only uses by only one method:
https://github.com/golangci/golangci-lint/blob/2a2eebdbea18a191837c1603dbc3fda59361da0e/pkg/lint/lintersdb/validator.go#L41-L48

Inside my PR, the method is now a function.

I think I will just remove the function `allPresetsSet()` (map) and use `AllPresets()` (slice), we don't need extreme performance optimization to browse 10 presets :smile:.